templates/default/kill-process.sh.erb

```diff
@@ -20,14 +20,19 @@ if [ ! -f "$PID_FILE"  ] ; then
 fi
 
 PID=$(cat "$PID_FILE")
-if [ "$INC_PID" -eq 1 ] ; then
-  echo "Incremeting the PID by 1 to skip over watchdog process"
-  PID=$((PID + 1))
+if [ "$ALL_PID" -eq 1 ] ; then
+# ndbmtd service uses this one
+# We need to perform killall -TERM ndbmtd to kill all ndbmtd processes
+  echo "Killing $NAME with killall -TERM $NAME"
+  (killall -TERM "$NAME") 2> /dev/null
+  RES=$?
+  wait_pid_removed=30
+else
+  echo "Killing $NAME with process-id $PID "
+  (kill -TERM "$PID") 2> /dev/null
+  RES=$?
+  wait_pid_removed=10
 fi
-echo "Killing $NAME with process-id $PID "
-(kill -TERM "$PID") 2> /dev/null
-RES=$?
-wait_pid_removed=10
 timeout=0
 while [ $timeout -lt $wait_pid_removed ] ; do
     sleep 1
```

Comment:
Should we not have a condition for whether you are killing all or just killing by ID?

if [ "$ALL_PID" -eq 1 ] ; then
 (killall -TERM "$NAME") 2>  && break
else
  (! kill -0 "$PID") 2> /dev/null && break
if